client/state/sharing/publicize/actions.js

```diff
@@ -59,14 +61,39 @@ export function sharePost( siteId, postId, skippedConnections, message ) {
  */
 export function fetchConnections( siteId ) {
 	return ( dispatch ) => {
-		dispatch( {
-			type: PUBLICIZE_CONNECTIONS_REQUEST,
-			siteId,
-		} );
+		dispatch( requestConnections( siteId ) );
 
 		return wpcom.undocumented().siteConnections( siteId )
-			.then( ( response ) => dispatch( receiveConnections( siteId, response ) ) )
-			.catch( ( error ) => dispatch( failConnectionsRequest( siteId, error ) ) );
+			.then( ( connections ) => {
+				dispatch( receiveConnections( siteId, connections ) );
+				dispatch( requestConnectionsSuccess( siteId ) );
+			} )
+			.catch( ( error ) => dispatch( requestConnectionsFailure( siteId, error ) ) );
+	};
+}
+
+/**
+ * Triggers a network request to request a Publicize connection for the
+ * specified site ID.
+ *
+ * @param  {Number} siteId       Site ID
+ * @param  {Number} connectionId ID of the connection to be fetched.
+ * @return {Function}            Action thunk
+ */
+export function fetchConnection( siteId, connectionId ) {
+	return ( dispatch ) => {
+		dispatch( requestConnections( siteId ) );
```

Comment:
Could we at least pass the connectionId through to the requestConnections()? This is a somewhat far off concern, but If we switch to api middleware or want to replay that action we'd be missing information.